dftimewolf/lib/processors/openrelik.py

```diff
@@ -0,0 +1,174 @@
+    """
+    super().__init__(state=state, name=name, critical=critical)
+    self.openrelik_api_client: api_client.APIClient = None
+    self.openrelik_folder_client: folders.FoldersAPI = None
+    self.openrelik_workflow_client: workflows.WorkflowsAPI = None
+    self.openrelik_api = None
+    self.openrelik_ui = None
+    self.openrelik_api_key = None
+    self.workflow_id = None
+    self.folder_id = None
+    self.sketch_id = None
+    self.incident_id = None
+
+  # pylint: disable=arguments-differ
+  def SetUp(
+    self,
+    incident_id: str | None,
+    sketch_id: int | None,
+    folder_id: int | None,
+    workflow_id: int | None,
+    openrelik_api: str | None,
+    openrelik_ui: str | None,
+    openrelik_api_key: str | None,
+  ) -> None:
+    self.openrelik_api = openrelik_api
+    self.openrelik_ui = openrelik_ui
+    self.openrelik_api_key = openrelik_api_key
+    self.openrelik_api_client = api_client.APIClient(
+      self.openrelik_api, self.openrelik_api_key
+    )
+    self.openrelik_folder_client = folders.FoldersAPI(self.openrelik_api_client)
+    self.openrelik_workflow_client = workflows.WorkflowsAPI(
+      self.openrelik_api_client
+    )
+
+    self.folder_id = folder_id
+    self.workflow_id = workflow_id
+    self.incident_id = incident_id
+    self.sketch_id = sketch_id
+
+  def PollWorkflowStatus(self, workflow_id: int) -> Iterator[str]:
+    """Polls the status of a workflow until it completes."""
+    filename = str(workflow_id)
+    workflow = self.openrelik_workflow_client.get_workflow(
+      self.folder_id, workflow_id
+    )
+    status = workflow.get("tasks")[0].get("status_short")
+    output_file_ids = []
+    while status != "SUCCESS" and status != "FAILED":
+      self.logger.info(f"Workflow {workflow_id} status: {status}")
+      time.sleep(5)
+      workflow = self.openrelik_workflow_client.get_workflow(
+        self.folder_id, workflow_id
+      )
+      status = workflow.get("tasks")[0].get("status_short")
+    self.logger.info(f"Workflow {workflow_id} status: {status}")
+    if status == "FAILED":
+      self.ModuleError(f"Workflow {workflow_id} failed", critical=True)
+
+    for task in workflow.get("tasks"):
+      output_files = task.get("output_files", [])
+      for output_file in output_files:
+        output_file_id = output_file.get("id")
+        filename = output_file.get("display_name", workflow_id)
+        output_file_ids.append(output_file_id)
+
+    for output_file_d in output_file_ids:
+      local_path = self.DownloadWorkflowOutput(output_file_d, filename)
+      yield local_path
+
+  def DownloadWorkflowOutput(self, file_id: int, filename: str) -> str:
+    """Downloads a file from OpenRelik.
+
+    Args:
+        file_id: The ID of the file to download.
+        filename: The name of the file to download.
+
+    Returns:
+        str: The path to the downloaded file.
+    """
+    endpoint = f"{self.openrelik_api_client.base_url}/files/{file_id}/download"
```

Comment:
Maybe add a logging.info call to indicate that we're downloading a file? since it can take a long time, we don't want users to think this is stuck

Reply:
Done